include/deal.II/grid/tria.h

```diff
@@ -2183,23 +2182,147 @@ class Triangulation : public Subscriptor
      * enum for more information). The function is expected to return an
      * unsigned integer, which is interpreted as the additional computational
-     * load of this cell. If this cell is going to be coarsened, the signal is
-     * called for the parent cell and you need to provide the weight of the
-     * future parent cell. If this cell is going to be refined the function
-     * should return a weight, which will be equally assigned to every future
-     * child cell of the current cell. As a reference a value of 1000 is added
-     * for every cell to the total weight. This means a signal return value of
-     * 1000 (resulting in a weight of 2000) means that it is twice as
-     * expensive for a process to handle this particular cell. If several
-     * functions are connected to this signal, their return values will be
-     * summed to calculate the final weight.
+     * load of this cell.
      *
-     * This function is used in step-68.
+     * In serial and parallel shared applications, partitioning happens after
+     * refinement. So all cells will have the `CELL_PERSIST` status.
+     *
+     * In parallel distributed applications, partitioning happens during
+     * refinement. If this cell is going to be coarsened, the signal is called
+     * for the parent cell and you need to provide the weight of the future
+     * parent cell. If this cell is going to be refined, the function is called
+     * on all children and you should ideally return the same weight for all
+     * children.
+     *
+     * If several functions are connected to this signal, their return values
+     * will be summed to calculate the final weight via `CellWeightSum`.
+     *
+     * This function is used in step-68 and implicitely in step-75 using the
+     * parallel::CellWeights class.
      */
     boost::signals2::signal<unsigned int(const cell_iterator &,
                                          const CellStatus),
                             CellWeightSum<unsigned int>>
-      cell_weight;
+      weight;
+
+#ifndef DOXYGEN
+    /**
+     * Legacy constructor that connects deprecated signals to their new ones.
+     */
+    Signals()
+      : cell_weight(weight)
+    {}
+
+    /**
+     * Legacy signal emulation to deprecate the old signal.
+     */
+    class LegacySignal
+    {
+    public:
+      using signature_type = unsigned int(const cell_iterator &,
+                                          const CellStatus);
+      using combiner_type  = CellWeightSum<unsigned int>;
+
+      using slot_function_type = boost::function<signature_type>;
+      using slot_type =
+        boost::signals2::slot<signature_type, slot_function_type>;
+
+      LegacySignal(
+        boost::signals2::signal<signature_type, combiner_type> &new_signal)
+        : new_signal(new_signal)
+      {}
+
+      ~LegacySignal()
+      {
+        base_weight.disconnect();
+      }
+
+      DEAL_II_DEPRECATED_EARLY
+      boost::signals2::connection
+      connect(
```

Comment:
I realize now that you have the discussion about deprecation in the discussion of the `cell_weight` variable below. Feel free to ignore this comment if you think that your solution is better.

Reply:
The documentation of the `cell_weigh` variable looks like [this](https://www.dealii.org/developer/doxygen/deal.II/structTriangulation_1_1Signals.html#a835640e3966ee5e0e41aad0302e5aaec). I find it descriptive enough for our deprecation purposes.

Of course, we can provide a more thorough documentation for this Legacy class, but as it will be removed anyways soon I wanted to not spend too much time on it.

On the other hand, the few minutes extra in writing documentation will help users who encounter problems with the deprecation. I will write a few more lines.

Reply:
That's fine -- let's not worry about it.